src/d3a/d3a_core/sim_results/area_statistics.py

```diff
@@ -132,49 +133,61 @@ def _is_prosumer_node(area):
     return isinstance(area.strategy, StorageStrategy)
 
 
-def _accumulate_load_trades(load, grid, accumulated_trades, is_cell_tower):
+def _accumulate_load_trades(load, grid, accumulated_trades, is_cell_tower, past_market_types):
     if load.name not in accumulated_trades:
         accumulated_trades[load.name] = {
             "type": "cell_tower" if is_cell_tower else "load",
-            "id": load.area_id,
             "produced": 0.0,
             "earned": 0.0,
             "consumedFrom": {},
             "spentTo": {},
         }
-    for market in grid.past_markets:
-        for trade in market.trades:
-            if trade.buyer == load.name:
-                sell_id = area_name_from_area_or_iaa_name(trade.seller)
-                accumulated_trades[load.name]["consumedFrom"] = add_or_create_key(
-                    accumulated_trades[load.name]["consumedFrom"], sell_id, trade.offer.energy)
-                accumulated_trades[load.name]["spentTo"] = add_or_create_key(
-                    accumulated_trades[load.name]["spentTo"], sell_id, trade.offer.price)
-    return accumulated_trades
 
+    markets = getattr(grid, past_market_types)
+    if markets is None:
+        return accumulated_trades
+    else:
+        if type(markets) != list:
+            markets = [markets]
+        for market in markets:
+            for trade in market.trades:
+                if trade.buyer == load.name:
+                    sell_id = area_name_from_area_or_iaa_name(trade.seller)
+                    accumulated_trades[load.name]["consumedFrom"] = add_or_create_key(
+                        accumulated_trades[load.name]["consumedFrom"], sell_id, trade.offer.energy)
+                    accumulated_trades[load.name]["spentTo"] = add_or_create_key(
+                        accumulated_trades[load.name]["spentTo"], sell_id, trade.offer.price)
+        return accumulated_trades
+
+
+def _accumulate_producer_trades(producer, grid, accumulated_trades, past_market_types):
+    if producer.name not in accumulated_trades:
```

Comment:
I think we discussed it offline too, but why does the assert need to be converted to an if ?

Reply:
Since, accumulated trades are recalculated if KEEP_PAST_MARKETS flag is set to True, Im deleting the old results at ever update_stats
https://github.com/gridsingularity/d3a/pull/457/files#diff-5e8e25677e3663098aa5bdd932bd1f8cR124